core/modules/field/field.module

```diff
@@ -1307,3 +1307,115 @@ function field_get_block_list() {
 
   return $types;
 }
+
+/**
+ * Implements hook_entity_view_mode_insert().
+ */
+function field_entity_view_mode_insert($view_mode, $entity_type) {
+  field_entity_view_mode_update($view_mode, $entity_type);
+}
+
+/**
+ * Implements hook_entity_view_mode_rename().
+ */
+function field_entity_view_mode_rename($entity_type, $view_mode_old, $view_mode_new) {
+  $entity_info = entity_get_info($entity_type);
+
+  foreach (array_keys($entity_info['bundles']) as $bundle) {
+    // Update all field bundle settings for the renamed view mode.
+    $settings = field_bundle_settings($entity_type, $bundle);
+    $changed = FALSE;
+    if (isset($settings['view_modes'][$view_mode_old])) {
+      $settings['view_modes'][$view_mode_new] = $settings['view_modes'][$view_mode_old];
+      unset($settings['view_modes'][$view_mode_old]);
+      $changed = TRUE;
+    }
+    if (isset($settings['extra_fields']['display'][$view_mode_old])) {
+      $settings['extra_fields']['display'][$view_mode_new] = $settings['extra_fields']['display'][$view_mode_old];
+      unset($settings['extra_fields']['display'][$view_mode_old]);
+      $changed = TRUE;
+    }
+    if ($changed) {
+      field_bundle_settings($entity_type, $bundle, $settings);
+    }
+
+    // Update all field instance display settings for the renamed view mode.
+    $instances = field_read_instances(array('entity_type' => $entity_type, 'bundle' => $bundle));
+    foreach ($instances as $instance) {
+      // Remove the view mode settings from all configured field instances.
+      if (isset($instance['display'][$view_mode_old])) {
+        $instance['display'][$view_mode_new] = $instance['display'][$view_mode_old];
+        field_update_instance($instance);
+      }
+    }
+  }
+}
+
+/**
+ * Implements hook_entity_view_mode_update().
+ */
+function field_entity_view_mode_update($view_mode, $entity_type) {
+  module_load_include('inc', 'field_ui', 'field_ui.admin');
+  $entity_info = entity_get_info($entity_type);
+  $view_mode_name = $view_mode['machine_name'];
+
+  // If custom settings is set to false, then we should be disabling this view
+  // view mode for all bundles.
+  //if (empty($view_mode['custom settings'])) {
+  //  $view_mode['enabled_bundles'] = array_fill_keys(array_keys($entity_info['bundles']), 0);
+  //}
```

Comment:
It should be removed, custom settings is not saved as part of the view mode in Backdrop, it's saved with field settings.